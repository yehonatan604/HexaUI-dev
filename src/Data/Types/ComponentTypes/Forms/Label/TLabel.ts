import { TComponent, ComponentOptions } from "../../TComponent";

export type TLabel = TComponent<"label"> & {
    options?: ComponentOptions;
    text: string;
}
