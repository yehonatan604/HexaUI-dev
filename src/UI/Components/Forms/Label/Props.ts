import { TComponent } from "../../../../Data/Types/TComponent";

export type LabelProps = TComponent<"label"> & {
    text: string;
};