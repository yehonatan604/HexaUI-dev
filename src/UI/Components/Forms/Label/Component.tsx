import useTheme from "../../../../Core/Hooks/useTheme";
import { TLabel } from "../../../../Data/Types/ComponentTypes/Forms/Label/TLabel";

const Label = (props: TLabel) => {
  const { text, options, ...componentProps } = props;

  // Hooks
  const { mode, colors } = useTheme();

  // Options
  const textColor =
    options?.textVariant ||
    (mode === "dark" ? colors.standradLight : colors.standradDark);

  // JSX
  return (
    <label className={`block text-sm font-medium text-${textColor}`} {...componentProps}>
      {text}
    </label>
  );
};

export default Label;
