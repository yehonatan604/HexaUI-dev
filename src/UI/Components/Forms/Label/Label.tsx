import { TLabel } from "./types/TLabel";

const Label = (props: TLabel) => {
  const { text, options, ...componentProps } = props;

  // Options
  const textColor = options?.textVariant
    ? `text-${options.textVariant}`
    : "dark:text-standard-l text-standard-d";

  // JSX
  return (
    <label className={`block text-sm font-medium ${textColor}`} {...componentProps}>
      {text}
    </label>
  );
};

export default Label;
