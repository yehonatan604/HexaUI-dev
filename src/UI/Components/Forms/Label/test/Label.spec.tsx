import { render, screen } from "@testing-library/react";
import Label from "../Label"; // Adjust the import path
import { describe, it, expect } from "vitest";

describe("Label Component", () => {
  it("should render the label with the provided text", () => {
    render(<Label text="Username" />);

    // Check if the label is rendered with the correct text
    expect(screen.getByText("Username")).toBeInTheDocument();
  });

  it("should apply the default CSS classes when no options are provided", () => {
    render(<Label text="Username" />);

    const label = screen.getByText("Username");

    // Check if the default text color and classes are applied
    expect(label).toHaveClass(
      "block",
      "text-sm",
      "font-medium",
      "dark:text-standard-l",
      "text-standard-d"
    );
  });

  it("should apply custom text color when textVariant option is provided", () => {
    render(<Label text="Password" options={{ textVariant: "blue" }} />);

    const label = screen.getByText("Password");

    // Check if the custom text color class is applied
    expect(label).toHaveClass("text-blue");
  });

  it("should render and match the snapshot", () => {
    const { asFragment } = render(<Label text="Snapshot Test" />);

    // Snapshot the label component structure
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render with custom textVariant and match the snapshot", () => {
    const { asFragment } = render(
      <Label text="Custom Snapshot" options={{ textVariant: "green" }} />
    );

    // Snapshot the label component structure with custom textVariant
    expect(asFragment()).toMatchSnapshot();
  });
});
