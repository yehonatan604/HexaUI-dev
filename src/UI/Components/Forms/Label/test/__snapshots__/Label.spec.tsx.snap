// Vitest Snapshot v1

exports[`Label Component > should render and match the snapshot 1`] = `
<DocumentFragment>
  <label
    class="block text-sm font-medium dark:text-standard-l text-standard-d"
  >
    Snapshot Test
  </label>
</DocumentFragment>
`;

exports[`Label Component > should render with custom textVariant and match the snapshot 1`] = `
<DocumentFragment>
  <label
    class="block text-sm font-medium text-green"
  >
    Custom Snapshot
  </label>
</DocumentFragment>
`;
